orangecontrib/imageanalytics/widgets/owimagegrid.py

```diff
@@ -328,27 +331,35 @@ def set_pixmap(future, thumb=thumbnail):
                 self.update_selection()
 
     def handleNewSignals(self):
+        self.data_is_subset = self._update_data_is_subset()
+        self.apply_subset()
+
+    def _update_data_is_subset(self):
         self.Warning.incompatible_subset.clear()
-        self.subset_indices = []
+        self.Warning.extras_in_subset.clear()
+        if not (self.data and self.data_subset):
+            return None
 
-        if self.data and self.data_subset and len(self.stringAttrs) > 0:
-            transformed = self.data_subset.transform(self.data.domain)
-            attr = self.stringAttrs[self.imageAttr]
+        subset_ids = set(self.data_subset.ids)
+        data_is_subset = np.array(
+            [id_ in subset_ids for id_ in self.data.ids if id_ in self.data.ids])
```

Comment:
The second part `if id_ in self.data.ids` does not seem like needed. :)

Reply:
Better safe than sorry.